### PR TITLE
feat(tabs): close tabs to the right and other tabs

### LINKED ISSUE #1133: Chrome like  Close tabs to the right / Close other tabs options
### What problem does this solve?

When many tabs are open, cleaning up clutter currently requires closing tabs one at a time via the X button or Cmd+W. Bulk-close actions — "Close Tabs to the Right" and "Close Other Tabs" — are a long-standing convention in browsers (Chrome, Firefox) and editors (VS Code, Sublime) for quickly pruning tabs without repetitive clicking.

### Proposed solution

Add two entries to the existing tab context menu (`src/modules/tabs/TabBar.tsx`):

- **Close Tabs to the Right** — closes every tab to the right of the clicked tab, scoped to the current space. Disabled when the clicked tab is already the rightmost.
- **Close Other Tabs** — closes every tab except the clicked tab, scoped to the current space. Disabled when only one tab is open.

Both actions:
- Reuse the existing dirty-editor and foreground-process close guards (one shared confirmation dialog, worded per action) so no unsaved work or running process is silently lost.
- Dispose the underlying terminal sessions for any tab that gets closed, same as a normal close.
- Compute the close set and next active tab id via pure, tested planner functions (`planCloseTabsToRight`, `planCloseOtherTabs`).

### Alternatives considered

- Multi-select + bulk close via modifier-click: more powerful, but a bigger UX surface than this ask needs.
- Keyboard-only shortcuts: doesn't fix the discoverability gap; the context menu is where users already look for this.

### Are you willing to contribute the implementation?

Yes, I can do it

### Before submitting

- [x] I searched existing issues and didn't find a duplicate

### PR DESCRIPTION
**Pull request by sohammondal**
## What

Closes #1133

Adds Chrome-style "Close tabs to the right" and "Close other tabs" actions to the tab context menu, with confirmation dialogs for dirty editors and terminals running a live process.

## Why

Tab strip lacked bulk-close actions; closing tabs one at a time is tedious. Both actions scope to the tab's space so cross-space tabs survive.

## How

- `planCloseTabsToRight` / `planCloseOtherTabs` in `useTabs.ts` compute which tab ids close (and the next active id) within the anchor's space; `closeTabsToRight` / `closeOtherTabs` apply them and dispose terminal sessions.
- `useTabCloseGuards` was refactored into a shared `handleCloseMany` / `pendingCloseMany` / `confirmCloseMany` path so both bulk actions reuse the same dirty-editor + foreground-process guard and confirmation dialog (title/wording keyed on a kind: "right" | "other").
- `Close other tabs` disabled when the space has a single tab; `Close tabs to the right` disabled when the tab is the rightmost.

## Testing

- [x] `pnpm lint` clean
- [x] `pnpm check-types` clean
- [x] `pnpm test` clean
- [x] Manual smoke-test of the affected feature
- [ ] (If you touched `src-tauri/`) `cargo clippy --all-targets --locked -- -D warnings` clean
- [ ] (If you touched `src-tauri/`) `cargo nextest run --locked` clean (or `cargo test --locked`)
- [ ] (If you changed a `#[tauri::command]` signature) called out below so the FE caller can be updated in lockstep
- [x] (If UI) tested in `pnpm tauri dev`
- [x] Platforms tested: macOS
- [ ] Shells tested (if relevant): 

## Screenshots / GIFs

https://github.com/user-attachments/assets/6979061c-a3c9-4b60-9c03-fb708f7168a8

https://github.com/user-attachments/assets/bae815b2-019f-48bc-b4fd-2f4a30d4bc4d

https://github.com/user-attachments/assets/459ffa3e-8196-48ad-8c96-a2f8e760f7ee

<kbd>Confirmation Dialog when multiple tabs have unsaved changes
Confirmation Dialog when multiple tabs have unsaved chages</kdb>

<kbd>Close options disabled when there are no tabs to closeClose options disabled when there are no tabs to close</kbd>

<kbd>Close to right option disable when there are no tabs to the rightClose to right option disable when there are no tabs to the right</kbd>

## Notes for reviewer

N/A

## Summary by CodeRabbit

- **New Features**
  - Added context-menu actions to close all tabs to the right or close all other tabs.
  - Added confirmation dialogs for bulk closures involving unsaved changes, running processes, or multiple tabs.
  - Automatically handles affected editor and terminal sessions while preserving an appropriate active tab.

- **Bug Fixes**
  - Improved bulk tab-closing behavior across tab spaces, including edge cases and unknown selections.

- **Tests**
  - Added coverage for bulk-close planning, hazard detection, session disposal, and active-tab fallback scenarios.

### CODE REVIEW

File: src/modules/tabs/lib/useTabs.ts [hunk 1]
```diff
@@ -1056,6 +1098,36 @@ export function useTabs(initial?: Partial<TerminalTab>) {
...
+      if (plan.closeIds.length === 0) return curr;
+      const close = new Set(plan.closeIds);
+      toDispose = curr
+        .filter((t) => close.has(t.id) && t.kind === "terminal")
+        .flatMap((t) => leafIds((t as TerminalTab).paneTree));
+      setActiveId(plan.nextActiveId);
+      return curr.filter((t) => !close.has(t.id));
+    });
+    for (const lid of toDispose) disposeSession(lid);
+  }, []);
```
[the remark this answers is not in the document]
Comment (sohammondal) [reply src/modules/tabs/lib/useTabs.ts]: Good catch on the purity concern, but I'm going to leave this as-is for this PR.

`closeTabsToRight`/`closeOtherTabs` intentionally mirror the existing `closeTab` (lines 1085-1099), which has the exact same shape: compute `toDispose` inside the `setTabs` updater closure, then call `disposeSession` after, outside the updater. That pattern predates this PR and isn't something I introduced net-new — I matched the established convention for `closeTab`'s siblings rather than diverging one of three near-identical functions.

Agreed it'd be worth hardening against replayed/deferred updaters. I'd rather not scope-creep this PR into an unrelated refactor, but if you'd rather have it addressed now instead of as a follow-up, I can update all three functions (`closeTab`, `closeTabsToRight`, `closeOtherTabs`) together in this same PR — just say the word.

**Comment by crynta** [comment]
Thank you for the work and great previews. Just added a couple of improvements and merged it!
